src/connectors/connectorKAF.js

```diff
@@ -0,0 +1,322 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Connector from "./connector";
+import { AS, Path } from "../model";
+import brembo, { parse } from "brembo";
+import ipUtils from "ip-sub";
+
+export default class ConnectorKAF extends Connector {
+
+    constructor(name, params, env) {
+        super(name, params, env);
+        this.consumer = null;
+        this.subscription = null;
+        this.agent = env.agent;
+        this.subscribed = {};
+
+        this.url = brembo.build(this.params.url, {
+            path: [],
+            params: {
+                client: env.clientId
+            }
+        });
+    };
+
+    _openConnect = (resolve) => {
+        resolve(true);
+
+        this._connect(this.name + ' connector connected');        
+
+        if (this.subscription) {
+            this.subscribe(this.subscription);
+        }
+    };
+
+    _messageToJson = (message) => {
+        this._message(JSON.parse(message));
+    };
+
+    _appendListeners = (resolve, reject) => {
+        this.consumer.on('message', (message) => {
+            this._message(message);
+            //console.log(message);
+        });
+        this.consumer.on('error', (error) => {
+            console.log("There was an error, disconnecting from broker")
+            if (this.connected) {
+                this._disconnect("RIPE RIS disconnected");
+            } else {
+                this._disconnect("It was not possible to establish a connection with RIPE RIS");
+                reject();
+            }
+            console.error(error);
+            this._error
+        });
+        this.consumer.on('connect', () => {
+            this._openConnect(resolve);
+        });
+    };
+
+    connect = () =>
+        new Promise((resolve, reject) => {
+            try {
+                var kafka = require('kafka-node');
+                var consumerOptions = {
+                    kafkaHost: this.params.broker,
+                    groupId: this.params.groupId,
+                    sessionTimeout: 15000,
+                    protocol: ['roundrobin'],
+                    fromOffset: 'latest'
+                }
+                //console.log(consumerOptions);
+                var topic = [this.params.topic];
+
+                var consumerGroup = new kafka.ConsumerGroup(Object.assign({ id: 'BGPAlerter' }, consumerOptions), topic);
+                this.consumer = consumerGroup;
+                this._appendListeners(resolve, reject);
+
+
+            } catch(error) {
+                this._error(error);
+                console.log(error);
+                reject(error);
+            }
+        });
+
+    disconnect = () => {
+        if (this.ws) {
+            this._disconnect(`${this.name} disconnected`);
+        }
+    };
+
+    _subscribeToAll = (input) => {

```

Comment:
This method must be implemented or the `carefulSubscription` option should be removed